docs/build/message-history.md

```diff
@@ -0,0 +1,136 @@
+---
+sidebar_label: Message history
+sidebar_position: 5.1
+description: Learn how to enable message history for group chat and why this feature is important to providing a seamless group chat user experience.
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Enable message history for group chat with XMTP
+
+Each app installation generates unique group chat keys. This means that group chat messages sent from one app installation can only be decrypted by that same app installation.
+
+To enable users to access their group chat messages using all of their app installations, enable the message history feature. Enabling message history is necessary for a seamless multi-app installation experience.
+
+To learn more about the user experience without the message history feature, see [⚠️ Important: Manage actions that make a local database inaccessible](/docs/build/group-chat/#️-important-manage-actions-that-make-a-local-database-inaccessible).
+
+## Run a message history server
+
+To enable the message history feature, you must run your own [**xmtp-message-history-server**](https://github.com/xmtp/xmtp-message-history-server). When you do, it will provide a message history sync URL to use in the next step.
+
+## Set message history sync URL
+
+When your app creates a new client, specify the location where the client should sync message history.
+
+<Tabs groupId="sdk-langs">
+<TabItem value="rn" label="React Native" attributes={{className: "rn_tab"}}>
+
+```jsx
+  const clientOptions = {
+    env: 'production' as const,
+    enableV3: true,
+    appVersion: 'YourApp/1.0.0',
+    // Add the history sync URL option
+    historySyncUrl: 'SYNC_URL'
```

Comment:
Maybe we should set our message history sync url here? Or should the sdk by default set it to the ephemera one? @tuddman did we decide if the SDK is going to set a message history url by default if they don't provide one?

Reply:
Love the idea of setting it to the Ephemera message history server URL by default... This way devs who aren't implementing message history won't have to do anything to ensure that they support the UX of other apps.

If anything, in the docs we can just disclose that this URL is set by default and if you don't want to participate - you can remove it.

However - is there any harm in having an app's message history collected on this server without their explicit permission?

Reply:
There's really two options to set here:
1. should message-history-sync be enabled (by default)?
2. and if so (enabled), what URL should the app use?  And if not set, what is our default?

We decided that option-1 -  should be the case in the SDK.   Message history sync ability- should be enabled by default.

> However - is there any harm in having an app's message history collected on this server without their explicit permission?

An app's message history is NOT collected on any server without their explicit permission.  Even if it were, its all encrypted so it's safe in that respect, but it's not stored without their explicit permission, so a non-issue.   The nuance is that new-device explicitly requests a history-sync, and it is the responding device with the same inbox-id that responds automatically.   But as the inbox owner is in all cases presumptively the same person, it's really the case that they are providing explicit permission.

Ephemera's History server URL is [http://history.dev.xmtp.network/](http://history.dev.xmtp.network/ )
We can spin up an additional server which will be accessible at [http://history.production.xmtp.network/](http://history.production.xmtp.network/) but the latter is not live yet.

Reply:
Thank you for this, @tuddman and @nplasterer! I have just a few comments and questions, please...

- Is message history sync ability enabled by default if the `historySyncUrl` needs to be set manually by the dev? The functionality is there - but it isn't enabled until the dev sets this value, is that correct? If we were to set the `historySyncUrl` value to http://history.dev.xmtp.network/ without dev intervention, I think this would truly be enabled by default?

- Regarding Ephemera's history server URL http://history.dev.xmtp.network/ - I wonder if we should adjust the URL to make it clear that it is a centralized server run by Ephemera?
   - Maybe history.dev.ephemera-xmtp.network? 

- Regarding the future URL http://history.production.xmtp.network/ - I don't think it needs to include "production" and can be just http://history.ephemera-xmtp.network/, for example.

Reply:
I think it makes sense for it to have the name ephemera in it! @tuddman can you make that change?

The ability to get the messages into another installation is enabled by default if we include a default url. But requesting message history is not enabled by default if that makes sense. The app you are using still needs to make a request for a message history but with this default url being set it atleast gives us a better chance of an app having a message history to sync from.

Reply:
Love it, @nplasterer - absolutely understood.

Providing a default message history server URL helps ensure that App A can provide message history for other apps to use without any manual steps required for the dev of App A.